cpusupport/cpusupport.h

```diff
@@ -94,6 +94,25 @@
 	int								\
 	cpusupport_ ## arch ## _ ## feature ## _detect_1(void)
 
+/**
+ * CPUSUPPORT_CHECK_USE(success_value, cpusupport_checks, check):
+ * Check if we can enable ${success_value}, given the ${cpusupport_checks} and
+ * ${check}.  If the ${cpusupport_checks} pass but the ${check} is non-zero,
+ * produce a warning which includes a stringified ${success_value}, then
+ * fallthrough.  This macro requires an 'hwaccel' variable.
+ */
+#define CPUSUPPORT_CHECK_USE(success_value, cpusupport_checks, check) do { \
+		if ((cpusupport_checks)) {				\
+			if ((check) == 0) {				\
+				hwaccel = (success_value);		\
```

Comment:
I would make `hwaccel` a macro parameter.  I know we're always using the same variable name, but it's ugly to have implied variables in a macro.